examples/Acoustics/compute_spectrum_production.py

```diff
@@ -0,0 +1,280 @@
+
+    if observer_id == -1:
+        print("Computing aeroacoustic output for all observers")
+    else:
+        print("Computing aeroacoustic output for observer ID " + str(observer_id))
+        if observer_id >= len(acoustics_data) - 2:
+            print("Observer ID not present in the aeroacoustics output")
+            raise RuntimeError
+
+    if window == "Hann":
+        print("Hann Window")
+    elif window is None:
+        print("No Window")
+    else:
+        print("Invalid windowing option selected")
+        raise RuntimeError
+
+    if weighting == "A":
+        print("A-weighted output")
+    elif weighting is None:
+        print("Non-weighted output")
+    else:
+        print("Invalid weighting option selected")
+        raise RuntimeError
+
+
+def acoustics_output(input_file, output, window, weighting, observer_id):
+    """This function drives the acoustics output"""
+
+    oaspl_all = []
+    observer_id_all = []
+    os.makedirs("figures", exist_ok=True)
+
+    acoustics_data = read_csv_file(input_file)
+    print_input_choices(observer_id, window, weighting, acoustics_data)
+
+    if observer_id == -1:
+        n_observer = len(acoustics_data) - 2
+        observer_list = list(range(0, n_observer))
+    else:
+        observer_list = list(range(observer_id, observer_id + 1))
+
+    if weighting == "A":
+        with open(output, "w", encoding="utf-8") as f:
+            f.write("observer_id, dBA \n")
+
+    elif weighting is None:
+        with open(output, "w", encoding="utf-8") as f:
+            f.write("observer_id, dB \n")
+
+    for i_observer in observer_list:
+        print(f"Observer ID = {i_observer}")
+        # Assuming constant time step size for now
+        time_step_size = acoustics_data["time"][1] - acoustics_data["time"][0]
+        # Apply windowing function (even if no windowing is specified, then rectangular window applied)
+        windowed_pressure = apply_windowing_function(acoustics_data, i_observer, window)
+        # Compute spectrum and OASPL values
+        frequency, spl, oaspl = compute_spectrum(
+            windowed_pressure, time_step_size, weighting
+        )
+        if np.isnan(oaspl) == 0:
+            observer_id_all = np.append(observer_id_all, i_observer)
+            oaspl_all = np.append(oaspl_all, oaspl)
+
+        # Output CSV file containing OASPL values
+        if np.isnan(oaspl) == 0:
+            with open(output, "a", encoding="utf-8") as f:
+                f.write(f"{i_observer}, {oaspl:.5f} \n")
+        else:
+            oaspl = 0
+            with open(output, "a", encoding="utf-8") as f:
+                f.write(f"{i_observer}, {oaspl:.5f} \n")
+
+        # Plot spectral response
+        if (np.isnan(spl).any()) == 0:
+            plot_spectrum(frequency, spl, i_observer, weighting, window)
+    plot_oaspl(observer_id_all, oaspl_all, weighting, window)
+
+
+INPUT_FILE = "09caab3a-dc97-4f40-9727-0fe94818cdf1_total_acoustics_v3.csv"
+acoustics_output(
```

Comment:
```suggestion
main(
```

Reply:
sorry, I actually see now that the original logic was something like the following, which I guess is a bit redundant. My understanding is that we will want to retain the `if __name__ == '__main__':` type structure to allow for importing these scripts as modules for future automations (eg, application-focused feature testing). Any recommendations here?

`
def main():
    input = 'file_path'
    acoustics_output(input)

if __name__ == '__main__':
    main()
`

Reply:
According to this:
https://www.notion.so/flexcompute/Flow360Scripts-31419058b846497c976af586695335d7 

We decided to remove main()

Reply:
think I confused the point previously, sorry. Resolving this and will add a clarification.